openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -641,9 +642,34 @@ def password_change_request_handler(request):
     else:
         return HttpResponseBadRequest(_("No email address provided."))
 
+def _get_rate(rate):
+    """
+    Given the request rate string, return a two tuple of:
+    <allowed number of requests>, <period of time in seconds>
+    """
 
-class PasswordResetTokenValidation(APIView):  # lint-amnesty, pylint: disable=missing-class-docstring
+    requests, duration = rate.split('/')
+    num_requests = int(requests)
+    num = int(duration[:-1] if duration[:-1] else 1)
+    symbol = duration[-1:]
+    duration = {'s': 1, 'm': 60, 'h': 3600, 'd': 86400}[symbol] * num
+    return (num_requests, duration)
```

Comment:
Use default formats or shift to `ratelimt` package. No need for this customization if there are already other options available.